### ISSUE TITLE
Calendar not working after update to owncloud 10

### ISSUE THREAD

**Issue by quenenni** [owncloud/calendar#795]
After owncloud update, the calendar app (and contact too) weren't available in my owncloud menu.
Via the 'Market' page (horrible page compared to what it exists before, where the F|@#| do you disable / enable an app?!), the install button for these apps just don't do  anything.
The error in the owncloud.log:
```
{"reqId":"qMQQlr20J1PmByUFwfwN","level":2,"time":"2017-05-02T14:31:03+02:00","remoteAddr":"94.177.226.137","user":"--","app":"core","method":"GET","url":"\/a2billing\/common\/javascript\/misc.js","message":"Trusted domain error. \"94.177.226.137\" tried to access using \"xxx.xxx.xxx.xxx\" as host."}
```

So, I had to download it manually, placed it in the app folder.
Next time on my owncloud, it told me it had to update the owncloud -> Success
Reload, but still no calendar app in my menu.

I had to do a "sudo -u www-data -s ./occ app:enable calendar" and only then it became available in my menu.

On the calendar page itself, all the layouts shows themselves correctly, but no calendars.
The error in the log:
```
{"reqId":"PuxvCB1QWC0Cz07ZbxSg","level":4,"time":"2017-05-02T17:13:13+02:00","remoteAddr":"xxx.xxx.xxx.xxx","user":"Aaaa","app":"webdav","method":"PROPFIND","url":"\/remote.php\/dav\/calendars\/Aaaa\/","message":"Exception: {\"Message\":\"An exception occurred while executing 'SELECT `publicuri` FROM `oc_dav_shares` WHERE (`resourceid` = ?) AND (`access` = ?)' with params [\\\"1\\\", 4]:\\n\\nSQLSTATE[42S22]: Column not found: 1054 Unknown column 'publicuri' in 'field list'\",\"Exception\":\"Doctrine\\\\DBAL\\\\Exception\\\\InvalidFieldNameException\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/DBALException.php(128): Doctrine\\\\DBAL\\\\Driver\\\\AbstractMySQLDriver->convertException('An exception oc...', Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOException))\\n#1 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Connection.php(855): Doctrine\\\\DBAL\\\\DBALException::driverExceptionDuringQuery(Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOMySql\\\\Driver), Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOException), 'SELECT `publicu...', Array)\\n#2 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/DB\\\/Connection.php(190): Doctrine\\\\DBAL\\\\Connection->executeQuery('SELECT `publicu...', Array, Array, NULL)\\n#3 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Query\\\/QueryBuilder.php(206): OC\\\\DB\\\\Connection->executeQuery('SELECT `publicu...', Array, Array)\\n#4 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/DB\\\/QueryBuilder\\\/QueryBuilder.php(141): Doctrine\\\\DBAL\\\\Query\\\\QueryBuilder->execute()\\n#5 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/CalDAV\\\/CalDavBackend.php(1644): OC\\\\DB\\\\QueryBuilder\\\\QueryBuilder->execute()\\n#6 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/CalDAV\\\/Calendar.php(267): OCA\\\\DAV\\\\CalDAV\\\\CalDavBackend->getPublishStatus(Object(OCA\\\\DAV\\\\CalDAV\\\\Calendar))\\n#7 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/CalDAV\\\/Publishing\\\/PublishPlugin.php(117): OCA\\\\DAV\\\\CalDAV\\\\Calendar->getPublishStatus()\\n#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/PropFind.php(98): OCA\\\\DAV\\\\CalDAV\\\\Publishing\\\\PublishPlugin->OCA\\\\DAV\\\\CalDAV\\\\Publishing\\\\{closure}()\\n#9 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/CalDAV\\\/Publishing\\\/PublishPlugin.php(124): Sabre\\\\DAV\\\\PropFind->handle('{http:\\\/\\\/calenda...', Object(Closure))\\n#10 [internal function]: OCA\\\\DAV\\\\CalDAV\\\\Publishing\\\\PublishPlugin->propFind(Object(Sabre\\\\DAV\\\\PropFind), Object(OCA\\\\DAV\\\\CalDAV\\\\Calendar))\\n#11 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#12 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1058): Sabre\\\\Event\\\\EventEmitter->emit('propFind', Array)\\n#13 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(980): Sabre\\\\DAV\\\\Server->getPropertiesByNode(Object(Sabre\\\\DAV\\\\PropFind), Object(OCA\\\\DAV\\\\CalDAV\\\\Calendar))\\n#14 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1665): Sabre\\\\DAV\\\\Server::getPropertiesIteratorForPath('calendars\\\/Aaaa', Array, 1)\\n#15 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(355): Sabre\\\\DAV\\\\Server->generateMultiStatus(Object(Generator), false)\\n#16 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPropFind(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#17 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#18 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(479): Sabre\\\\Event\\\\EventEmitter->emit('method:PROPFIND', Array)\\n#19 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#20 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Server.php(226): Sabre\\\\DAV\\\\Server->exec()\\n#21 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(31): OCA\\\\DAV\\\\Server->exec()\\n#22 \\\/var\\\/www\\\/owncloud\\\/remote.php(165): require_once('\\\/var\\\/www\\\/ownclo...')\\n#23 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Driver\\\/AbstractMySQLDriver.php\",\"Line\":71,\"User\":\"Aaaa\"}"}
```
And indeed, there is no "publicuri" column in the dav_shares table.

**Comment by quenenni** [comment]
I found this ticket from Core that resolves the publicuri problem: https://github.com/owncloud/core/issues/27754

The other problem, the old remote.php/caldav/... address, that should work even if it's /dav/ now), just doesn't work (error 404)
But that's an old bug that I already resolved by closing thunderbird and executing a small script that changes the /caldav/ to /dav/ in the TB settings.js. (I thought it was a problem from my config, but it seems no)

**Comment by DeepDiver1975** [comment]
will be fixed in owncloud/core#27754
